src/main/query_result.cpp

```diff
@@ -5,12 +5,65 @@
 #include "binder/expression/node_rel_expression.h"
 #include "binder/expression/property_expression.h"
 
-using namespace std;
 using namespace kuzu::processor;
 
 namespace kuzu {
 namespace main {
 
+std::unique_ptr<DataTypeInfo> DataTypeInfo::getInfoForDataType(
+    const DataType& type, const string& name) {
+    auto columnTypeInfo = make_unique<DataTypeInfo>(type.typeID, name);
+    switch (type.typeID) {
+    case common::NODE_ID: {
+        columnTypeInfo->childrenTypesInfo.push_back(
+            make_unique<DataTypeInfo>(common::INT64, "offset"));
+        columnTypeInfo->childrenTypesInfo.push_back(
+            make_unique<DataTypeInfo>(common::INT64, "tableID"));
+    } break;
+    case common::LIST: {
+        auto parentTypeInfo = columnTypeInfo.get();
+        auto childType = type.childType.get();
+        parentTypeInfo->childrenTypesInfo.push_back(getInfoForDataType(*childType, ""));
+    } break;
+    default: {
+        // DO NOTHING
+    }
+    }
+    return std::move(columnTypeInfo);
+}
+
+vector<unique_ptr<DataTypeInfo>> QueryResult::getColumnTypesInfo() {
+    vector<unique_ptr<DataTypeInfo>> result;
+    for (auto i = 0u; i < columnDataTypes.size(); i++) {
+        auto columnTypeInfo = DataTypeInfo::getInfoForDataType(columnDataTypes[i], columnNames[i]);
+        if (columnTypeInfo->typeID == common::NODE || columnTypeInfo->typeID == common::REL) {
```

Comment:
Maybe this outer `if` can be removed and the `else` can be changed to `else if (columnTypeInfo->typeID == common::REL)`

Reply:
Right. Done.